### DISCUSSION TITLE
4 Semester 3 Woche

### DISCUSSION THREAD

**Discussion by MelonigeMelone** [MelonigeMelone/xCalendar#28] [Blogkanal]
# Refactoring

## Offene Pull Requests

Diese Woche haben wir den Code und unser Repository etwas aufgeräumt. Im ersten Schritt haben wir erst einmal noch offene Pull Requests im Frontend abgeschossen und unseren Main Branch auf den neusten Stand gebracht. Die alten "Feature Branches" wurden gelöscht.
Der GraphQl Branch wird weiterhin als separater Branch existieren, da dieser noch etwas experimentell gehalten ist und noch nicht funktionsfähig ist.

## Dokumente zum Git Repository hinzugefügt

Wir haben angefangen unsere Dokumente in GitHub hochzuladen. Müssen diese allerdings noch einzeln zusammentragen, sodass die Vollständigkeit noch nicht garantiert ist und manche Dateien noch als PDF hochgeladen sind und durch entsprechende Word Dateien etc. ersetzt werden im Laufe der nächsten Tage.

## Code Refactoring
Unser Hauptmerkmal lag im ersten Schritt darin den Code übersichtlicher zu gestalten, vor allem durch das Optimieren von den Imports, sowie den Refraktoren vom Code Format.
Zusätzlich wurden teilweise doppelte oder falsch gesetzte CSS Attribute entfernt und fehlende Tags bei HTML Elementen hinzugefügt. Dabei ging es lediglich um den Code Stil.
In einem nächsten Schritt sollen kleiner Fehler im CSS behoben werden, um die Darstellung des Frontend zu verbessern.

**Comment by YPuest** [comment]
Guten Tag zusammen, eure im Blogeintrag erwähnten Anpassungen und Änderungen wirken logisch und durchdacht. Ich persönlich würde mich bei der Refaktorierung über konkrete Beispiele im Blog freuen.
Ansonsten freue ich mich schon auf euren nächsten Eintrag :)

**Comment by Christian-2003** [comment]
Hallo zusammen,
es erscheint mir, als hättet ihr diese Woche viel zu tun gehabt. Wenn ihr damit diese Woche weiter macht, dann ist euer Repository bestimmt auf dem neuesten Stand und ihr könnt effektiv weiter programmieren.

Ich freue mich schon auf die nächste Woche.
